src/fsharp/vs/ServiceUntypedParse.fs

```diff
@@ -726,13 +730,99 @@ module internal UntypedParseInfoImpl =
                     Some CompletionContext.Invalid
             | _ -> None
 
+        let (|Operator|_|) name e = 
```

Comment:
@enricosada - We don't use "function" in this way much in the F# comiler codebase, again for purposes of readability, and also to be able to set debug breakpoints.  In general the use of "function" is de-emphasised in F# coding when defining a named function (e.g. we hardly use it at all in Expert F# 3.0).  

Indeed for the F# compiler codebase I would consider it good code cleanup to remove a use of  

    let someFunction x = function ...

in favour of 

    let someFunction x y =
        match y with ...

because you can give "y" a good, documented name, set a breakpoint on the "match" and the code is just easier to read and understand.